pkg/framework/test/fixtures.go

```diff
@@ -0,0 +1,83 @@
+package test
```

Comment:
A couple of random thoughts we just discussed:

- We tried to make the interface for the processes we start in our framework rather generic. That should make it relatively easy to introduce new processes (Scheduler, Controller(Manager), ...) somewhen in the future.
- We see that the APIServer depends on Etcd -- from that point of view it might be a good idea to make this dependency explicit as you suggested.
- I believe we can run the APIServer without Etcd. When Etcd eventually comes up APIServer will connect and everything will just work. We might have test which actually test with a missing Etcd and/or an dis- and reappearing Etcd.
- ( What if I'd like to run my tests against a already running instance of Etcd on a different machine or so?[1] )

So we are not against making the dependency explicit, we think both ways have their pros and cons. I -- personally -- would opt for keeping it as is for now, but that's just a gut feeling. We are happy to discuss that in more depth!


[1] That just came to our minds -- we know that is not supported or even planned to be supported by our framework. However we thought that might be a valuable use case to keep in mind ...